docs-src/concepts/what-is-worker-versioning.md

```diff
@@ -0,0 +1,187 @@
+day to complete, you know that you'll never have to leave older Workers running for more than a day
+after you deploy the new version (assuming availability). You don't need to worry about removing old
+versions from the sets. Leaving them there is harmless.
+
+You can use this technique with longer-lived Workflows as well. Just be aware that you might need to
+keep multiple versions of Workers running while open Workflows complete.
+
+You can also use the feature to prevent a buggy codepath from being taken on currently open
+Workflows. You can do this by adding a new version to an existing set and specifying it as
+_compatible_ with an existing version on which no future Workflow Tasks should execute. Because the
+new version processes existing [Event Histories](/workflows/#event-history), it is subject to the
+normal [deterministic constraints](/workflows/#deterministic-constraints), and you might need to use
+one of the [versioning APIs](/workflows/#workflow-versioning).
+
+The feature also permits you to make incompatible changes to Activity Definitions in conjunction
+with incompatible changes to Workflow Definitions that use those Activities. This works because any
+Activity scheduled by a Workfow on the same Task Queue will, by default, only be dispatched to
+Workers that have a compatible build ID with the Workflow that scheduled it. Thus, if you wish
+to change the type signature of an Activity Definition while creating a new incompatible build ID
+for a Worker, you can do so without concern that the Activity will fail to execute on some other
+Worker with an incompatible definition. The same principle applies to Child Workflows.
+
+Note that if you schedule an Activity or a Child Workflow on _different_ Task Queue from the one
+the Workflow is running on, they will be not be assigned any particular version.
+
+### How to use Worker Versioning
+
+To use this feature, follow these steps:
+
+1. Define Worker build-identifier version sets for the Task Queue.
+   You can use either the `temporal` CLI or your choice of SDK.
+   **_TODO: Links_**
+2. Enable the feature on your Worker by specifying a build identifier.
+   **_TODO: Links for each SDK's Worker options_**
+
+#### Defining the version sets
+
+Whether you use `temporal` cli or an SDK, updating the version sets feels the same. You specify the
+Task Queue that you're targeting, the build identifier that you're adding (or promoting), whether it
+becomes the new default version, and any existing versions it should be considered compatible with.
+
+The rest of this section uses updates to one Task Queue's version sets as examples.
+
+A Task Queue starts its life in an unversioned state. Workers who have not opted into the feature
+receive Tasks as usual. Workers who have opted in don't poll the Task Queue because they can't
+receive Tasks for Workflows that have no associated version.
+
+:::note Versions don't need to be semver!
```

Comment:
nit: it's not semver if it's only 2 parts.

Reply:
nit on your nit: 2-part version numbers can still be in the general class of semantic versions, even if they're not "semver".
```suggestion
:::note Versions don't need to follow semver or any other semantic versioning standard!
```